.github/workflows/rust.yml

```diff
@@ -65,42 +65,71 @@ jobs:
           # this key equals the ones on `linux-build-lib` for re-use
           key: cargo-cache-benchmark-${{ hashFiles('datafusion/**/Cargo.toml', 'benchmarks/Cargo.toml', 'datafusion-cli/Cargo.toml') }}
 
-      - name: Check workspace without default features
-        run: cargo check --no-default-features -p datafusion
+      - name: Check datafusion without default features
+        run: cargo check --all-targets --no-default-features -p datafusion
 
       - name: Check datafusion-common without default features
-        run: cargo check --tests --no-default-features -p datafusion-common
+        run: cargo check --all-targets --no-default-features -p datafusion-common
+
+      - name: Check datafusion-functions
+        run: cargo check --all-targets --no-default-features -p datafusion-functions
 
       - name: Check workspace in debug mode
         run: cargo check
 
       - name: Check workspace with all features
         run: cargo check --workspace --benches --features avro,json
 
+      - name: Check Cargo.lock for datafusion-cli
```

Comment:
Since this test often fails on PRs, I moved it closer to the start of this CI check to fail more quickly